### ISSUE TITLE
Build failure with CLI beta.22-1

### ISSUE THREAD

**Issue by buddyackerman** [angular/angular-cli#3575]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
> Windows 10

### Versions.
CLI version beta 22-1
NG version 2.2.4
Typescript version 2.0.10

### Repro steps.
Just doing **ng build**

### The log given by the failure.

Hash: d3b335d11433de5fdbd4
Time: 15718ms
chunk    {0} main.bundle.js, main.bundle.map (main) 16.7 kB {3} [initial] [rendered]
chunk    {1} scripts.bundle.js, scripts.bundle.map (scripts) 363 kB {4} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.map (styles) 10.1 kB {4} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.map (vendor) 3.1 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.map (inline) 0 bytes [entry] [rendered]

ERROR in ./src/app/app.module.ts
Module build failed: Error: Debug Failure. False expression: Declaration file is not present only for javascript files
    at Object.assert (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:2437:23)
    at getDeclFileName (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:37246:26)
    at onSingleFileEmit (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:7569:13)
    at Object.forEachExpectedEmitFile (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:7542:21)
    at writeReferencePath (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:37233:20)
    at C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:35750:29
    at Object.forEach (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:1115:30)
    at C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:35742:20
    at Object.forEach (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:1115:30)
    at emitDeclarations (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:35735:12)
    at getDeclarationDiagnosticsFromFile (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:35699:13)
    at onSingleFileEmit (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:7569:13)
    at Object.forEachExpectedEmitFile (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:7542:21)
    at Object.getDeclarationDiagnostics (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:35695:12)
    at C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:45673:27
    at runWithCancellationToken (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:45468:24)
    at getDeclarationDiagnosticsWorker (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:45669:20)
    at getDeclarationDiagnosticsForFile (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:45677:60)
    at getDiagnosticsHelper (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:45436:24)
    at Object.getDeclarationDiagnostics (C:\Projects\Angular2\node_modules\typescript\lib\typescript.js:45460:24)
    at TypeScriptFileRefactor.getDiagnostics (C:\Projects\Angular2\node_modules\@ngtools\webpack\src\refactor.js:51:35)
    at _checkDiagnostics (C:\Projects\Angular2\node_modules\@ngtools\webpack\src\loader.js:106:32)
    at C:\Projects\Angular2\node_modules\@ngtools\webpack\src\loader.js:140:17
 @ ./src/app/index.ts 2:0-29
 @ ./src/main.ts
 @ multi main

### Mention any other details that might be useful.
This error comes after the build is complete and the dist files are created.

This actually started when I was trying to build with aot and after reading several issues and articles I decided to update to that latest CLI version and the latest compatible NG version and then this started happening, even when I wasn't building with aot enabled.  I just want understand what the error is saying so I know where to look to fix it.

**Comment by buddyackerman** [comment]
Update:  I downgraded the CLI to beta.21 and downgraded NG to version 2.2.1 and I now only get the above error when trying to build with **aot** set to true.

**Comment by hansl** [comment]
TL;DR: Some library you're using is not compatible with the CLI.

You're using a library that was not deployed properly. Basically when building TypeScript there are `d.ts` files built with typing information. That library was deployed without those `.d.ts` files. The CLI requires this typing information and we require libraries to be AoT compatible for working seamlessly with the CLI. Contact the vendor (or tell us the library name and we can contact them) and let them know, or change to a different library.

We're working towards making the error easier to understand.

Closing this as a dupe and as WAI.

**Comment by buddyackerman** [comment]
To test your theory I created a new project using the CLI.  I then installed the same packages that I have in my "failing" project and added the imports to the app.module and then added some simple uses of those libraries to the app.component.  It built in AoT mode just fine and I was able to load the app in my browser.  I then replaced the app folder of this new project with the app folder of the failing project and built it and it failed with the same error as above.

**Comment by hansl** [comment]
Could you list the libraries in question?

**Comment by buddyackerman** [comment]
Here's my package.json from the new test project that I setup.  



    {
      "name": "ng2-cli-test4",
      "version": "0.0.0",
      "license": "MIT",
      "angular-cli": {},
      "scripts": {
        "start": "ng serve",
        "lint": "tslint \"src/**/*.ts\"",
        "test": "ng test",
        "pree2e": "webdriver-manager update",
        "e2e": "protractor"
      },
      "private": true,
      "dependencies": {
        "@angular/common": "2.2.1",
        "@angular/compiler": "2.2.1",
        "@angular/core": "2.2.1",
        "@angular/forms": "2.2.1",
        "@angular/http": "2.2.1",
        "@angular/platform-browser": "2.2.1",
        "@angular/platform-browser-dynamic": "2.2.1",
        "@angular/router": "3.2.1",
        "core-js": "^2.4.1",
        "lodash": "^4.17.2",
        "ng2-bootstrap": "^1.1.16",
        "primeng": "^1.1.0",
        "rxjs": "5.0.0-beta.12",
        "ts-helpers": "^1.1.1",
        "zone.js": "^0.6.23"
      },
      "devDependencies": {
        "@angular/compiler-cli": "2.2.1",
        "@types/jasmine": "2.5.38",
        "@types/lodash": "^4.14.43",
        "@types/node": "^6.0.42",
        "angular-cli": "1.0.0-beta.21",
        "codelyzer": "~1.0.0-beta.3",
        "jasmine-core": "2.5.2",
        "jasmine-spec-reporter": "2.5.0",
        "jquery": "^3.1.1",
        "karma": "1.2.0",
        "karma-chrome-launcher": "^2.0.0",
        "karma-cli": "^1.0.1",
        "karma-jasmine": "^1.0.2",
        "karma-remap-istanbul": "^0.2.1",
        "protractor": "4.0.9",
        "tether": "^1.4.0",
        "ts-node": "1.2.1",
        "tslint": "3.13.0",
        "typescript": "~2.0.3",
        "webdriver-manager": "10.2.5"
      }
    }

**Comment by hansl** [comment]
You don't have typings for jquery.

`ng2-bootstrap` seems to have problems with AOT.

**Comment by buddyackerman** [comment]
I added the types for jquery and removed ng2-bootstrap from the project and I get that same error.

**Comment by buddyackerman** [comment]
I've removed all third party libraries from my project and it still fails to build with AoT .  My current configuration is ng version 2.2.1 and ng-cli version beta.21 and below are my current dependencies from package.json:



    "dependencies": {
        "@angular/common": "^2.2.1",
        "@angular/compiler": "^2.2.1",
        "@angular/core": "^2.2.1",
        "@angular/forms": "^2.2.1",
        "@angular/http": "^2.2.1",
        "@angular/platform-browser": "^2.2.1",
        "@angular/platform-browser-dynamic": "^2.2.1",
        "@angular/router": "^3.2.1",
        "core-js": "^2.4.1",
        "rxjs": "5.0.0-beta.12",
        "ts-helpers": "^1.1.1",
        "zone.js": "^0.6.23"
    }
